src/app/dim-ui/WeaponArchetype.ts

```diff
@@ -1,12 +1,14 @@
-import { DimItem } from 'app/inventory/item-types';
+import { DimItem, DimSocket, PluggableInventoryItemDefinition } from 'app/inventory/item-types';
 import { SocketCategoryHashes } from 'data/d2/generated-enums';
 
-export const getWeaponArchetypeSocket = (item: DimItem) =>
-  (item.bucket.inWeapons &&
-    !item.isExotic &&
-    item.sockets?.categories.find((c) => c.category.hash === SocketCategoryHashes.IntrinsicTraits)
-      ?.sockets[0]) ||
-  undefined;
+export function getWeaponArchetypeSocket(item: DimItem): DimSocket | undefined {
+  if (item.bucket.inWeapons && !item.isExotic) {
+    return item.sockets?.categories.find(
+      (c) => c.category.hash === SocketCategoryHashes.IntrinsicTraits
+    )?.sockets[0];
+  }
+}
 
-export const getWeaponArchetype = (item: DimItem) =>
-  getWeaponArchetypeSocket(item)?.plugged?.plugDef;
+export const getWeaponArchetype: (item: DimItem) => PluggableInventoryItemDefinition | undefined = (
```

Comment:
I still prefer this syntax, because it doesn't involve repeating the arguments list:

```ts
export const getWeaponArchetype = 
  (item: DimItem): PluggableInventoryItemDefinition | undefined => 
    getWeaponArchetypeSocket(item)?.plugged?.plugDef
```

Reply:
fair i am always just rearranging the equals sign and colon until it stops causing syntax errors